### PR TITLE
Implemented feature toggle service for FR

### PR DESCRIPTION
**Pull request by johnoneill46**
- Created Feature toggle service to enable us to have different toggles per env

- Added 2 toggles that although are not used right now (FEATURE_HWF_SUCCESSFUL_NOTIFICATION_LETTER & FEATURE_ASSIGNED_TO_JUDGE_NOTIFICATION_LETTER) will be used in a separate PR I've currently got waiting

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java [hunk 1]
```diff
@@ -88,7 +91,7 @@
                 .orderLetter(approvedConsentOrderLetter)
                 .consentOrder(consentOrderAnnexStamped);
 
-            if (approvedConsentOrderNotificationLetterFeature) {
+            if (featureToggleService.isFeatureEnabled(APPROVED_CONSENT_ORDER_NOTIFICATION_LETTER)) {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java:94]: can we please add more methods:

```featureToggleService.isApprovedConsentOrderNotificantionEnabled()```

In my opinion it looks better

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/impl/FeatureToggleServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+@ConfigurationProperties(prefix = "feature-toggle")
+@Configuration
+@Getter
+public class FeatureToggleServiceImpl implements FeatureToggleService {
```
Comment (shaunganley) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/impl/FeatureToggleServiceImpl.java:20]: might be a standard on this project already but I would name these classes: 
```suggestion
public class FeatureToggleService implements IFeatureToggleService {
```

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ConsentOrderApprovedDocumentServiceTest.java [hunk 3]
```diff
@@ -104,7 +104,7 @@ public void shouldGenerateApprovedConsentOrderLetter() {
 
     @Test
     public void shouldGenerateApprovedConsentOrderNotificationLetterForApplicant() {
-        if (!approvedConsentOrderNotificationLetterFeature) {
+        if (!featureToggleService.isFeatureEnabled(APPROVED_CONSENT_ORDER_NOTIFICATION_LETTER)) {
```
Comment (jvictorcf) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ConsentOrderApprovedDocumentServiceTest.java:107]: Sorry, I didn't quite understand the intent behind this test.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/FeatureToggleServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    private FeatureToggleServiceImpl classToTest = new FeatureToggleServiceImpl();
+
+    @Before
+    public void setup() {
+        Map<String, String> toggles = new HashMap<>();
+        toggles.put(Features.APPROVED_CONSENT_ORDER_NOTIFICATION_LETTER.getName(), "false");
+        toggles.put(Features.HWF_SUCCESSFUL_NOTIFICATION_LETTER.getName(), "true");
+
+        ReflectionTestUtils.setField(classToTest, "toggle", toggles);
```
Comment (jvictorcf) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/FeatureToggleServiceImplTest.java:25]: I'm not a big fan at all of using reflection for tests.
I'd rather have a constructor that takes the toggles as a parameter or something of the sort.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java [hunk 5]
```diff
@@ -31,8 +31,8 @@
     private static final String DOCUMENT_URL = "document_binary_url";
     private static final String VALUE = "value";
 
-    @Value("${feature.approved-consent-order-notification-letter}")
-    private boolean approvedConsentOrderNotificationLetterFeature;
+    @Autowired
```
Comment (jakub-kozlowski) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java:34]: `Autowired` is deprecated - where possible use `private final` for services and Lombok's `@RequiredArgsContructor` ? or is there a reason to leave it package proctected?

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -115,4 +115,7 @@ idam.s2s-auth.services-allowed-to-update=${BSP_SERVICES_ALLOWED_TO_UPDATE:bulk_s
 idam.s2s-auth.services-allowed-to-validate=${BSP_SERVICES_ALLOWED_TO_VALIDATE:bulk_scan_processor}
 
 # Features
-feature.approved-consent-order-notification-letter=${FEATURE_APPROVED_CONSENT_ORDER_NOTIFICATION_LETTER:false}
+feature-toggle.service.api.baseurl = ${FEATURE_TOGGLE_SERVICE_API_BASEURL:http://localhost:4028}
```
Comment (jvictorcf) [inline src/main/resources/application.properties:118]: I might have missed this, but how do you build these environment variables?

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintServiceTest.java [hunk 7]
```diff
@@ -33,8 +34,8 @@
     @Autowired
     private BulkPrintService bulkPrintService;
 
-    @Value("${feature.approved-consent-order-notification-letter}")
-    private boolean featureApprovedConsentOrderNotificationLetter;
+    @Autowired
```
Comment (jakub-kozlowski) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintServiceTest.java:37]: ditto - here it's not even package protected! :)

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java (hunk 1 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/ConsentOrderApprovedController.java]: Agreed - changes made 👍

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/impl/FeatureToggleServiceImpl.java (hunk 2 above)
Comment (johnoneill46) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/impl/FeatureToggleServiceImpl.java]: @shaunganley  Ended up getting rid of the interface entirely and just having a standalone service as it's unlikely we'll be requiring different implementations of the FeatureToggle service

**Pull request merged**